ion-schema/src/result.rs

```diff
@@ -91,3 +91,19 @@ pub fn unresolvable_schema_error_raw<S: AsRef<str>>(description: S) -> IonSchema
         description: description.as_ref().to_string(),
     }
 }
+
+/// A macro that checks some condition required to be valid ISL.
+///
+/// If invalid, returns an InvalidSchemaErr with the given error message.
+#[macro_export]
+macro_rules! isl_require {
+    ($expression:expr => $fmt_string:literal $(, $($tt:tt)*)?) => {
```

Comment:
🗺️ Nifty little macro to make it more concise to validate the ISL.

Example usage:
```rust
isl_require!(foo.is_bar() => "Foo must be bar, but was {foo}")?;
```

I'm not very keen on the `=>` to separate the condition from the format args, but there's a very limited set of characters that are allowed to follow an `expr` designator. (I originally wanted `else`, but the other options are `,` and `;`.)

Feel free to tell me that it should be `,` or `;` instead if you have an opinion on the matter.